packages/rate-limit/src/rate-limit.ts

```diff
@@ -74,7 +74,7 @@ export function rateLimit(options?: Partial<RateLimitOptions>) {
       const { current, resetTime } = await incrementStore(key)
       const maxResult = typeof max === 'function' ? await max(req, res) : max
 
-      // @ts-ignore         // todo adjust typing
+      // @ts-ignore
       req.rateLimit = {
```

Comment:
you can extend the tinyhttp's `Request` to have extra options

and export that interface in order to use it everywhere else